src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h

```diff
@@ -53,7 +53,8 @@ MPIDI_POSIX_eager_send(int grank, MPIDI_POSIX_am_header_t * msg_hdr, const void
     transport = MPIDI_POSIX_eager_iqueue_get_transport(src_vsi, dst_vsi);
 
     /* Try to get a new cell to hold the message */
-    MPIDU_genq_shmem_pool_cell_alloc(transport->cell_pool, (void **) &cell);
+    MPIDU_genq_shmem_pool_cell_alloc(transport->cell_pool, (void **) &cell,
+                                     MPIDI_POSIX_global.local_ranks[MPIR_Process.local_rank]);
```

Comment:
Do we need the parameter or we could simply do this internally as a global policy, e.g. CVAR

Reply:
I like having the parameter in case there's a reason to use the alternative way of doing things. This makes it easy to specify which pool things come out of.

As an added benefit, it made it easy for me to compare using a local cell vs. a remote cell when doing performance testing.

Reply:
There may also be times where it is more helpful to do a genq in one way for one part of the code and another way for another part of the code.

Reply:
This discussion should move to https://github.com/pmodels/mpich/pull/5759.